frb_rust/src/misc/logging.rs

```diff
@@ -0,0 +1,233 @@
+#[macro_export]
+macro_rules! enable_frb_logging {
+    () => {
+#[allow(clippy::crate_in_macro_def)]
+use crate::frb_generated::StreamSink;
+use flutter_rust_bridge::frb as frb_4_log;
+pub use log::{LevelFilter, Metadata, Record};
+
+#[frb(dart_code = "
+import 'dart:io';
+import 'package:logging/logging.dart';
+
+static void default_log_function(Log2DartLogRecord record) {
+  print(
+    '[${DateTime.now()} ${log_level_from_number(record.levelNumber)} @${record.rustLog ? 'Rust' : 'Dart'} ${record.loggerName}] ${record.message}');
+}
+
+/// initialize the logging system, including the rust logger
+static Logger init_logger(
```

Comment:
I changed the API, so that the configuration is simpler.

These settings need to happen at the same time.
The problem is this code:
```Rust
    pub fn initialize_log2dart(log_stream: crate::frb_generated::StreamSink<Log2DartLogRecord>, max_log_level: u16) {
      log::set_boxed_logger(Box::new(FRBLogger {
        stream_sink: log_stream,
      }))
      .map(|()| log::set_max_level(from_u16(max_log_level)))
      .expect("initialize_log2dart is called only once!");
(...)
```
Because I move the `FRBLogger` into `log::` I cannot change the log level subsequentially anymore. 
In between I had a version where one could set this and the logging function later in the program execution - but while I was struggling with lifetimes I thought that a user might never need to change these later in a program.

If it is just to make the function smaller and easier to read: Sure, we can split it up. But eventually it needs to be called at once.

Reply:
> But eventually it needs to be called at once.

By default yes. But users should be free to customize etc.

Suppose a simple example: User may like to have Rust log forward to Dart, but at the same time, he or she already setup the dart logging (e.g. his own Logger.root.level, his own Logger.root.onRecord, let's say he sends it to Sentry instead of printing for some reason). Then he should *not* call our 2nd function.

Reply:
> I cannot change the log level subsequentially anymore.

What about

```
pub fn another_function_that_freely_changes_log_level(){
  log::set_max_level(TRACE)
}
```

seems we can change the level

Reply:
Ok, I will look into it. I forgot why I wasn't able to change it after the initialization ... maybe it was a temporary limitation during refactoring ...

But let's first refactor where the code will be generated to, do unit tests - and then I will enable setting the log level after initialization.

As for your example, a user could create another logger for `Sentry`, that inherits from the FRB created root logger and change anything he likes there.

But in general, I think if a user changes something using this logging implementation, it should always apply to both sides, rust and dart. That way it is least surprising :)